src/Adapter/Filesystem/SatelliteBuilder.php

```diff
@@ -100,6 +100,11 @@ public function build(): Satellite\SatelliteInterface
             $composer,
         );
 
+        // TODO: allow adding $to
+        foreach ($this->paths as [$from, $to]) {
+            $satellite->withFile($from);
+        }
+
```

Comment:
```suggestion
        $satellite->withFile(...$this->files);

```

the `$paths` property has no use and should be removed from the class. The `$files` property is the one containing your `FileInterface` objects to transfer to the Satellite instance.